### ISSUE TITLE
Dropout op在downscale模式下不支持单元素tensor输入的推理

### ISSUE THREAD

**Issue by Kevin-XiongC** [PaddlePaddle/Paddle#49794]
### bug描述 Describe the Bug

``` python
import paddle
paddle.nn.functional.dropout(paddle.to_tensor([1.]),mode="downscale_in_infer",training=False)
```
运行报以下错
``` shell
 (InvalidArgument) Input dimension number(num_col_dims) must be between 0 and 1, but received number is 1.
  [Hint: Expected (num_col_dims > 0 && num_col_dims < rank) == true, but received (num_col_dims > 0 && num_col_dims < rank):0 != true:1.] (at /Users/paddle/xly/workspace/f2bafd01-b80e-4ac8-972c-1652775b2e51/Paddle/paddle/phi/kernels/funcs/eigen/common.h:107)
```
环境
Mac M1
paddle 2.4.1

**Comment by ronny1996** [comment]
好的，问题已经复现，我反馈一下

**Comment by ronny1996** [comment]
可以先这样使用绕一下，后续会修复
`paddle.nn.functional.dropout(paddle.to_tensor([[1.]]),mode="downscale_in_infer",training=False)`
